docs/developers/developer-reference/sc-contract-calls.md

```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
+              .execute_on_dest_context_readonly()
+              .execute_on_same_context()"| exec-dest["⚙️ Synchronous call"]
+    exec-dest --> result[Requested Result]
+```
+
+---
+
+[comment]: # (mx-context-auto)
+
+## Contract deploy: base
+
+A close relative of the contract call is the contract deploy call. It models the deployment or the upgrade of a smart contract.
+
+It shares a lot in common with the contract calls, with these notable differences:
+- The endpoint name is always `init`.
+- No ESDT transfers are allowed in `init`.
+- They get executed slightly differently.
+
+
+
+```mermaid
+flowchart TB
+    gen-proxy[Generated Proxy] --> cc
+    man-proxy[Manual Proxy] --> cc
+    ccnp-new["ContractDeploy::new"] --> cc
+    cc[ContractDeploy]
+    cc ---->|".deploy_contract()
+              .deploy_from_source()"| exec-deploy["⚙️ Deploy contract"]
+    exec-deploy --> result[Requested Result]
+    cc ---->|".deploy_contract()
+              .deploy_from_source()"| exec-upg["⚙️ Upgrade contract"]
+```
+
+The object encoding these calls is called `ContractDeploy`. Unlike the contract calls, there is a single such object.
+
+Creating this object is done in a similar fashion: either via proxies, or by hand. Constructors in proxies naturally produce `ContractDeploy` objects:
 
-Sometimes you don't have access to the callee contract code, or it's simply inconvenient to import it (different framework versions, for instance). In this case, you're going to have to manually declare your proxy. Let's use the same example endpoint as in the first method:
 ```rust
 mod callee_proxy {
     multiversx_sc::imports!();
 
     #[multiversx_sc::proxy]
     pub trait CalleeContract {
-        #[payable("*")]
-		#[endpoint(myEndpoint)]
-		fn my_payable_endpoint(&self, arg: BigUint) -> BigUint;
+		#[init]
+		fn init(&self, arg: BigUint) -> BigUint;
     }
 }
 ```
 
-This is the only thing you'll have to do differently. The rest is the same, you declare a proxy builder, and the calls are all exactly the same as in method #1.
 ```rust
-#[proxy]
-fn contract_proxy(&self, sc_address: ManagedAddress) -> callee_proxy::Proxy<Self::Api>;
+self.contract_proxy()
+	.init(my_biguint_arg)
 ```
 
+---
+
 [comment]: # (mx-context-auto)
 
-### Method #3: Manual calls (NOT recommended)
+## Contract deploy: configuration
 
-If for some reason you don't want to use the contract proxies, you can create a `ContractCall` object by hand:
+Just like with regular contract calls, we can specify the gas limit and perform EGLD transfers as part of the deploy as follows:
 
 ```rust
-let mut contract_call = ContractCall::new(
-    self.api,
-    dest_sc_address,
-    ManagedBuffer::new_from_bytes(endpoint_name),
-);
+self.contract_proxy(callee_sc_address)
+	.init(my_biguint_arg)
+
 ```
 
-Where `dest_sc_address` is the address of the callee contract, and `endpoint_name` would be `b"myEndpoint"`.
+```rust
+self.contract_proxy()
+	.init(my_biguint_arg)
+	.with_egld_transfer(egld_amount)
```

Comment:
is `with_egld_transfer` available on deploy? I think this is an upgrade, right?

Reply:
It is!